library/X509/Job.php

```diff
@@ -141,6 +165,66 @@ private static function generateTargets(ConfigObject $jobDescription, array $hos
         }
     }
 
+    protected function getScanTargets()
+    {
+        if (! $this->isRescan()) {
+            yield from $this->generateTargets();
+        } else {
+            $targets = (new Select())
+                ->columns(['id', 'ip', 'hostname', 'port'])
+                ->from('x509_target');
+
+            if ($this->getLastScan()) {
+                if ($this->db->getAdapter() instanceof Pgsql) {
+                    $targets->where(new Expression(
+                        sprintf('last_scan < NOW() - INTERVAL \'%d\'', $this->getLastScan())
+                    ));
+                } else {
+                    $targets->where(new Expression(sprintf('last_scan < NOW() - %d', $this->getLastScan())));
+                }
+            }
+
+            $targets = $this->db->select($targets);
+            $targets->setFetchMode(\PDO::FETCH_OBJ);
+
+            foreach ($targets as $target) {
+                $addr = gmp_import(DbTool::unmarshalBinary($target->ip));
+                $addrFound = false;
+
+                foreach ($this->getCidrs() as $cidr) {
+                    if (self::isAddrInside($cidr, $addr)) {
+                        list($subnet, $_) = $cidr;
+                        $target->ip = self::numberToAddr($addr, self::isIPV6($subnet)); // Replace binary IP
+                        $addrFound = true;
+
+                        break;
+                    }
+                }
+
+                if ($addrFound) {
+                    $this->updateLastScan($target);
```

Comment:
Based on the method name, I wouldn't expect the last scan to be updated here. And I would not do that either.